gruntfile.js

```diff
@@ -248,69 +259,229 @@ module.exports = function (grunt) {
+  grunt.task.registerTask(
+    'mkdir:upload',
+    'Task that makes sure upload directory exists.',
+    function() {
+      // Get the callback
+      var done = this.async();
 
-    grunt.file.mkdir(path.normalize(__dirname + '/modules/users/client/img/profile/uploads'));
+      grunt.file.mkdir(
+        path.normalize(__dirname + '/modules/users/client/img/profile/uploads')
+      );
 
-    done();
-  });
+      done();
+    }
+  );
 
   // Connect to the MongoDB instance and load the models
-  grunt.task.registerTask('mongoose', 'Task that connects to the MongoDB instance and loads the application models.', function () {
-    // Get the callback
-    var done = this.async();
+  grunt.task.registerTask(
+    'mongoose',
+    'Task that connects to the MongoDB instance and loads the application models.',
+    function() {
+      // Get the callback
+      var done = this.async();
 
-    // Use mongoose configuration
-    var mongoose = require('./config/lib/mongoose.js');
+      // Use mongoose configuration
+      var mongoose = require('./config/lib/mongoose.js');
 
-    // Connect to database
-    mongoose.connect(function (db) {
-      done();
-    });
-  });
+      // Connect to database
+      mongoose.connect(function(db) {
+        done();
+      });
+    }
+  );
 
-  grunt.task.registerTask('server', 'Starting the server', function () {
-    // Get the callback
-    var done = this.async();
+  /**
+   * Start the local functional test server.
+   */
+  var server_process;
+  grunt.task.registerTask(
+    'server',
+    'Starting functional test server.',
+    function() {
+      var done = this.async();
+
+      server_process = childProcess.spawn('node', ['server.js'], {
+        env: process.env,
+        detached: true,
+        shell: false,
+        stdio: 'ignore'
+      });
 
-    var path = require('path');
-    var app = require(path.resolve('./config/lib/app'));
-    var server = app.start(function () {
       done();
-    });
-  });
+    }
+  );
+
+  /**
+   * Run the functional tests against the local server/database.
+   */
+  grunt.task.registerTask(
+    'functional',
+    'Running functional tests.',
+    function() {
+      var done = this.async();
+
+      /**
+       * To run in headful mode:
+       * ['chromeTest']
+       *
+       * To run a single test:
+       * ['-DchromeTest.single=SomeSpecName', 'chromeTest']
+       *
+       * To increase logging (including println):
+       * ['chromeHeadlessTest', '--info']
+       */
+      var test_process = childProcess.spawn(
+        process.platform == 'win32' ? 'gradlew.bat' : './gradlew',
+        ['chromeHeadlessTest'],
+        {
+          env: process.env,
+          cwd: path.join(process.cwd(), 'functional-tests'),
+          stdio: 'inherit'
+        }
+      );
+
+      test_process.on('exit', done);
+    }
+  );
+
+  /**
+   * Drop the local functional test database.
+   */
+  grunt.task.registerTask(
+    'drop_database',
```

Comment:
Nit.: if this is only for the functional database, it may be worth naming it accordingly (e.g.: drop_functional_database).
Same for the kill server function that comes next.

Reply:
Good call. will change.